src/PrivateKey.cpp

```
@@ -137,191 +103,34 @@ PrivateKey::PrivateKey(
     append(bytes, extension2);
     append(bytes, chainCode2);
     _curve = curve;
+    auto* privkey = Rust::tw_private_key_create_with_data(bytes.data(), bytes.size(), static_cast<uint32_t>(curve));
+    _impl = privkey ? Rust::wrapTWPrivateKey(privkey) : nullptr;
```

Same, let's throw an exception